### PR TITLE
Split main service into dedicated ones

### PR DESCRIPTION
**Pull request by archendel**

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -771,15 +792,15 @@ public ResponseEntity<String> getNodeNetworkModifications(@Parameter(description
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The node report has been deleted"), @ApiResponse(responseCode = "404", description = "The study/node is not found")})
     public ResponseEntity<Void> deleteNodeReport(@Parameter(description = "Study uuid") @PathVariable("studyUuid") UUID studyUuid,
```
Comment (bendaamerahm) [inline src/main/java/org/gridsuite/study/server/StudyController.java:793]: same for studyUuid

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 2]
```diff
@@ -754,7 +775,7 @@ public ResponseEntity<Void> stopSecurityAnalysis(@Parameter(description = "Study
     public ResponseEntity<List<ReporterModel>> getNodeReport(@Parameter(description = "Study uuid") @PathVariable("studyUuid") UUID studyUuid,
                                                              @Parameter(description = "Node uuid") @PathVariable("nodeUuid") UUID nodeUuid,
                                                              @Parameter(description = "Node only report") @RequestParam(value = "nodeOnlyReport", required = false, defaultValue = "true") boolean nodeOnlyReport) {
-        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.getNodeReport(studyUuid, nodeUuid, nodeOnlyReport));
```
Comment (bendaamerahm) [inline src/main/java/org/gridsuite/study/server/StudyController.java:757]: remove studyUuid from PathVariable if we don't using it

**Review by bendaamerahm — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/service/NetworkConversionService.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
+package org.gridsuite.study.server.service;
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/service/NetworkConversionService.java:1]: Missing license header and author

File: src/main/java/org/gridsuite/study/server/service/NetworkMapService.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
+package org.gridsuite.study.server.service;
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/service/NetworkMapService.java:1]: Missing license header and author

File: src/main/java/org/gridsuite/study/server/service/SecurityAnalysisService.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
+package org.gridsuite.study.server.service;
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/service/SecurityAnalysisService.java:1]: Missing license header and author

File: src/main/java/org/gridsuite/study/server/service/SingleLineDiagramService.java [hunk 6]
```diff
@@ -0,0 +1,188 @@
+package org.gridsuite.study.server.service;
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/study/server/service/SingleLineDiagramService.java:1]: Missing license header and author

File: src/test/java/org/gridsuite/study/server/SecurityAnalysisTest.java [hunk 7]
```diff
@@ -0,0 +1,449 @@
...
+                studyUuid, nodeUuid, CONTINGENCY_LIST_NAME))
+                .andReturn();
+        resultAsString = mvcResult.getResponse().getContentAsString();
+        Integer integerResponse = Integer.parseInt(resultAsString);
+        assertEquals(integerResponse, Integer.valueOf(1));
+
+        assertTrue(TestUtils.getRequestsDone(1, server).stream().anyMatch(r -> r.matches("/v1/contingency-lists/" + CONTINGENCY_LIST_NAME + "/export\\?networkUuid=" + NETWORK_UUID_STRING + ".*")));
+    }
+
+    private StudyEntity insertDummyStudy(UUID networkUuid, UUID caseUuid) {
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/study/server/SecurityAnalysisTest.java:377]: Maybe all the utility functions like insertDummyStudy, getRootNode, createNetworkModificationNode ... used in all services test classes could be moved in TestUtils.java, in order to not duplicate them everywhere ...

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 1 above)
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/StudyController.java:793]: fixed

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 2 above)
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/StudyController.java:757]: Fixed

File: src/test/java/org/gridsuite/study/server/utils/TestUtils.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+                assertNull("Should not be any messages in queue " + destination + " : ", output.receive(TIMEOUT, destination));
+            });
+        } catch (NullPointerException e) {
+            // Ignoring
+        } finally {
+            output.clear(); // purge in order to not fail the other tests
+        }
+    }
+
+    public static void assertServerRequestsEmptyThenShutsown(MockWebServer server) throws UncheckedInterruptedException, IOException {
```
[the remark this answers is not in the document]
Comment (archendel) [reply src/test/java/org/gridsuite/study/server/utils/TestUtils.java:88]: fixed

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 9]
```diff
@@ -627,7 +648,7 @@ public ResponseEntity<String> getSecurityAnalysisResult(@Parameter(description =
                                                                   @Parameter(description = "nodeUuid") @PathVariable("nodeUuid") UUID nodeUuid,
                                                                   @Parameter(description = "Limit types") @RequestParam(name = "limitType", required = false) List<String> limitTypes) {
         List<String> nonNullLimitTypes = limitTypes != null ? limitTypes : Collections.emptyList();
-        String result = studyService.getSecurityAnalysisResult(nodeUuid, nonNullLimitTypes);
+        String result = securityAnalysisService.getSecurityAnalysisResult(nodeUuid, nonNullLimitTypes);
         return result != null ? ResponseEntity.ok().body(result) :
                ResponseEntity.noContent().build();
     }
```
[the remark this answers is not in the document]
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/StudyController.java:654]: fixed, moved stopSecurityAnalysis as well

File: src/main/java/org/gridsuite/study/server/service/ActionsService.java [hunk 10]
```diff
@@ -0,0 +1,55 @@
+package org.gridsuite.study.server.service;
```
[the remark this answers is not in the document]
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/service/ActionsService.java:1]: fixed

File: src/main/java/org/gridsuite/study/server/service/CaseService.java [hunk 11]
```diff
@@ -0,0 +1,129 @@
+package org.gridsuite.study.server.service;
```
[the remark this answers is not in the document]
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/service/CaseService.java:1]: fixed

File: src/main/java/org/gridsuite/study/server/service/ConsumerService.java [hunk 12]
```diff
@@ -0,0 +1,200 @@
+package org.gridsuite.study.server.service;
```
[the remark this answers is not in the document]
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/service/ConsumerService.java:1]: fixed

File: src/main/java/org/gridsuite/study/server/service/GeoDataService.java [hunk 13]
```diff
@@ -0,0 +1,63 @@
+package org.gridsuite.study.server.service;
```
[the remark this answers is not in the document]
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/service/GeoDataService.java:1]: fixed

File: src/main/java/org/gridsuite/study/server/service/LoadflowService.java [hunk 14]
```diff
@@ -0,0 +1,212 @@
+package org.gridsuite.study.server.service;
```
[the remark this answers is not in the document]
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/service/LoadflowService.java:1]: fixed

File: src/main/java/org/gridsuite/study/server/service/NetworkConversionService.java (hunk 3 above)
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/service/NetworkConversionService.java:1]: fixed

File: src/main/java/org/gridsuite/study/server/service/NetworkMapService.java (hunk 4 above)
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/service/NetworkMapService.java:1]: fixed

File: src/main/java/org/gridsuite/study/server/service/SecurityAnalysisService.java (hunk 5 above)
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/service/SecurityAnalysisService.java:1]: fixed

File: src/main/java/org/gridsuite/study/server/service/SingleLineDiagramService.java (hunk 6 above)
Comment (archendel) [reply src/main/java/org/gridsuite/study/server/service/SingleLineDiagramService.java:1]: fixed

**Review by FranckLecuyer — APPROVED** [review]

**Review by bendaamerahm — APPROVED** [review]

**Pull request merged**